.github/workflows/docker_publish.yml

```
@@ -26,192 +26,73 @@ jobs:
-          fi
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - uses: actions/checkout@v2
-      - name: log into GHCR, setup Docker Buildx, Cache Docker layers
-        uses: ./.github/actions/login-buildx-cache
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Build and push moab/dagmc
-        uses: docker/build-push-action@v2
-        with:
-          file: docker/ubuntu_20.04-dev.dockerfile
-          context: .
-          push: true
-          build-args: |
-            py_version=3.8
-            build_pyne=NO
-            build_moab=YES
-            enable_pymoab=${{ env.PYMOAB }}
-            build_dagmc=${{ env.DAGMC }}
-            build_hdf5=${{ env.HDF5 }}
-            install_openmc=${{ env.HDF5 }}
-          cache-from: type=local,src=/tmp/.buildx-cache-${{ github.sha }}
-          tags: ${{ env.DOCKER_IMAGE_BASENAME }}${{ matrix.hdf5 }}${{ matrix.dagmc }}${{ matrix.pymoab }}${{ matrix.openmc }}_pyne-deps:ci_testing
-
-  # Once the previous two steps have finished the docker images are uploaded
-  # to ghcr and therefore the local cached images are removed. 
-  cleanup_cache:
-    if: always()
-    needs: [build_and_push_python3_pymoab, build_and_push_dagmc]
-    runs-on: ubuntu-latest
-    steps:
-      - name: Move cache
-        run: |
-          rm -rf /tmp/.buildx-cache-${{ github.sha }}
+          repository: ${{ env.DOCKER_IMAGE_BASENAME }}/hdf5
+          stages: base_python, moab
+          server-stage: dagmc
+          quiet: false
+          dockerfile: docker/ubuntu_20.04-dev.dockerfile
+          build-args: py_version=3.8, build_hdf5=hdf5-1_12_0
 
   # Downloads the images uploaded to ghcr in previous stages and runs pyne
   # tests to check that they work.
   BuildTest:
-    needs: [build_and_push_python3_pymoab, build_and_push_dagmc]
+    needs: [multistage_image_build]
     runs-on: ubuntu-latest
 
-      
     strategy:
       matrix:
-        hdf5 : [
-          '',
-        ]
-        dagmc : [
-          '',
-          _dagmc
-        ]
-        pymoab : [
-          _pymoab,
-          ]
-        openmc : [
-          '',
-        ]
+        stage: [base_python, moab, dagmc, openmc]
+        hdf5: ['']
         include:
-          - hdf5: _hdf5-1_12_0
-            dagmc: _dagmc
-            pymoab: _pymoab
-            openmc: ''
-          - hdf5: ''
-            dagmc: _dagmc
-            pymoab: _pymoab
-            openmc: _openmc
-          - hdf5: ''
-            dagmc: ''
-            pymoab: ''
-            openmc: ''
+          - stage: dagmc
+            hdf5: /hdf5
       fail-fast: false
 
 
     container:
-      image: ghcr.io/${{ github.repository_owner }}/ubuntu_20.04_py3${{ matrix.hdf5 }}${{ matrix.dagmc }}${{ matrix.pymoab }}${{ matrix.openmc }}_pyne-deps:ci_testing
+      image: ghcr.io/${{ github.repository_owner }}/pyne/ubuntu_20.04_py3${{ matrix.hdf5 }}/${{ matrix.stage }}:refs_heads_multistage-docker-bk0
```

Can you use ${{ env.DOCKER_IMAGE_BASENAME }} here?  Also switch to underscore instead of directory-like naming
```suggestion
      image: ${{ env.DOCKER_IMAGE_BASENAME }}${{ matrix.hdf5 }}_${{ matrix.stage }}:refs_heads_multistage-docker-bk0
```

---

I think we don't use the env variable here because of an error we ran into in #1461 where the env variable couldn't be used because env isn't defined before matrix in the workflow run.